lib/controller.ts

```diff
@@ -0,0 +1,93 @@
+import { defaultEventNames } from './utils'
+
+type Context = {
+  [key: string]: Function
+}
+
+export default class Controller {
+  private dataAttr: string
+  private targetElements: NodeList
+  private actionElements: NodeList
+
+  private cache = {
```

Comment:
Cache might be a bad name here. Any suggestion on terms?

Reply:
Can't think of a better one. Good enough, I think.

But you should make things readonly if possible  :)

```suggestion
  private readonly cache = {
```

Reply:
Ok, will go with cache for this one.
In this case cache is not readonly (see the updateCache method). :)